### PR TITLE
feat: Add support for -fprofile-prefix-path gcc option (#1507)

### PR DESCRIPTION
**Pull request by jdavidpeter**
Current CCache (4.10.2) doesn't support shared caching for coverage builds in different build workspaces using gcc as the "apparent real path" of the object file is always part of the cache key.

This is needed as gcc by default uses the "apparent real path" of the object file to create gcda files and therefore these paths will be embedded in the object file.

gcc 10+ added the -fprofile-prefix-path=path option that can be used to lstrip the gcda filenames and enable the same object files to be created in different build workspaces when using --coverage.
This change adds support for  -fprofile-prefix-path=path thus enables ccache to reuse object files across build workspaces with the correct settings.

See https://github.com/ccache/ccache/issues/1507 for more details.

### CODE REVIEW

File: test/suites/profiling_gcc_10+.bash [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# Remove header, including a volatile timestamp, from a .gcno file.
```
Comment (jrosdahl) [inline test/suites/profiling_gcc_10+.bash:1]: The comment does not seem to describe the function?

**Review by jrosdahl — COMMENTED** [review]
Thanks for writing a test case, much appreciated!

File: test/suites/profiling_gcc_10+.bash [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    # -------------------------------------------------------------------------
+    TEST "-fprofile-prefirx-path=pwd, -fprofile-dir=., gcc coverage build"
+    # GCC 10 and newer allows to lstrip the mangled absolute path in the generated gcda file name
+    # This doesn't effect the absolute cwd path in the gcno file (but there's sloppiness for that)
+    CCACHE_SLOPPINESS_OLD="$CCACHE_SLOPPINESS"
+    export CCACHE_SLOPPINESS="$CCACHE_SLOPPINESS gcno_cwd"
+
+    export CCACHE_DEBUG=ON
+    export CCACHE_DEBUGDIR=/mnt/c/Temp/debug_ccache
```
[the remark this answers is not in the document]
Comment (jdavidpeter) [reply test/suites/profiling_gcc_10+.bash:64]: Will do right now, thank you for noting!

File: test/suites/profiling_gcc_10+.bash [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+SUITE_profiling_gcc_10+_SETUP() {
+    echo 'int main(void) { return 0; }' >test.c
+    unset CCACHE_NODIRECT
+}
+
+SUITE_profiling_gcc_10+() {
+    # -------------------------------------------------------------------------
+    TEST "-fprofile-prefirx-path=path, gcc coverage build"
+    # When using a rekative path for -fprofile-dir in gcc 9+, absolute object file path will be mangled in the .gcda filename
```
[the remark this answers is not in the document]
Comment (jdavidpeter) [reply test/suites/profiling_gcc_10+.bash:35]: Ah, the typo checking CI was so cool, I'm just a little sad it didn't catch everything.

File: test/suites/profiling_gcc_10+.bash (hunk 1 above)
Comment (jdavidpeter) [reply test/suites/profiling_gcc_10+.bash:1]: Copy-paste leftover, deleted.

File: src/ccache/argprocessing.cpp [hunk 4]
```diff
@@ -901,6 +901,14 @@ process_option_arg(const Context& ctx,
   if (util::starts_with(arg, "-fprofile-")
       || util::starts_with(arg, "-fauto-profile")
       || arg == "-fbranch-probabilities") {
+    if (util::starts_with(arg, "-fprofile-prefix-path=")) {
+      std::filesystem::path profile_prefix_path = arg.substr(arg.find('=') + 1);
+      args_info.profile_prefix_path = profile_prefix_path;
+      LOG("Set profile prefix path to {}", args_info.profile_prefix_path);
+      state.common_args.push_back(args[i]);
+      return Statistic::none;
+    }
```
[the remark this answers is not in the document]
Comment (jdavidpeter) [reply src/ccache/argprocessing.cpp:910]: Yes, it belongs there, this code reflects my earlier and wrong understanding of how compiler options are hashed to the cache key.

[the remark this answers is not in the document]
Comment (jdavidpeter) [reply src/ccache/argprocessing.cpp]: I have moved it, please check if it's good now.

**Comment by jrosdahl** [comment]
Thanks.

**Pull request merged**